Frontend/Gmfctn/src/app/modules/+auth/components/forgot-password/forgot-password.component.ts

```diff
@@ -0,0 +1,62 @@
+import { Component, OnInit, SimpleChanges } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { Router } from '@angular/router';
+import { SuccesfullyResetComponent } from 'src/app/shared/components/succesfully-reset/succesfully-reset.component';
+import { ResetPasswordService } from '../../services/reset-password/reset-password.service';
+
+@Component({
+  selector: 'app-forgot-password',
+  templateUrl: './forgot-password.component.html',
+  styleUrls: ['./forgot-password.component.scss']
+})
+export class ForgotPasswordComponent implements OnInit {
+  email = '';
+  resetPasswordForm = new FormGroup({});
+  dataToReset = {newPassword: '', hash: ''};
+  errorOccured = false;
+  isMessageSent = false;
+  emailRegexp = /^[^\s@]+@[^\s@]+$/;
+  isValid = false;
+
+  constructor(private readonly fb: FormBuilder, private resetPasswordService: ResetPasswordService,
+              private router: Router, private snackBar: MatSnackBar) { }
+
+  ngOnInit(): void {
+    document.querySelector('#mail')?.addEventListener('keydown', this.emailValidator.bind(this));
```

Comment:
Use reactive forms for this, accessing dom in components are bad practice

Reply:
Fixed